### PR TITLE
Frontend feat 2467 accessibility-purple ride ui changes and api integration

### PR DESCRIPTION
**Pull request by MercyQueen**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### CODE REVIEW

File: Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java [hunk 1]
```diff
@@ -1933,7 +1933,7 @@ public void horizontalScrollToPos(final String id, final String childId, final i
                 if (x ==0 && y == 0) {
                 scrollView.fullScroll(focus);
                 } else {
-                scrollView.scrollTo(x, y);
+                scrollView.smoothScrollTo(x, y);
```
Comment (kavya-shree-s) [inline Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java:1936]: @Vignesh-772 Why is scrollTo replaced with smoothScrollTo ?

Comment (kavya-shree-s) [reply Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java:1936]: @MercyQueen Can we revert back to scrollTo?

File: Frontend/ui-common/src/Components/SelectListModal/View.purs [hunk 2]
```diff
@@ -118,36 +129,31 @@ linearLayout
...
-    , padding (PaddingTop 4)
-    , gravity CENTER
+    , gravity config.subHeadingTextConfig.gravity 
     , color config.subHeadingTextConfig.color
     ] <> FontStyle.paragraphText TypoGraphy) 
     ,linearLayout
     [ width MATCH_PARENT
     , height WRAP_CONTENT
     , orientation VERTICAL
+    , padding (Padding 20 0 20 0)
```
Comment (kavya-shree-s) [inline Frontend/ui-common/src/Components/SelectListModal/View.purs:148]: use PaddingHorizontal

File: Frontend/ui-customer/src/Resources/Constants.purs [hunk 3]
```diff
@@ -297,3 +297,13 @@ getVehicleCapacity variant = case getMerchant FunctionCall of
...
           _ -> "4 " <> (getString SEATS)
   _ ->    ""
+
+getDisabilityType :: String -> Array ST.DisabilityT -> ST.DisabilityT 
+getDisabilityType disType disList = (fromMaybe dummyDisabilityList (head (filter(\item -> item.tag == disType) disList)))
+
+dummyDisabilityList :: ST.DisabilityT
+dummyDisabilityList ={
+  tag : "OTHER",
+  id : "8a365d73-b81e-6b21-962b-b1397aa687e0",
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Resources/Constants.purs:307]: @MercyQueen Why id is hardcoded

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs [hunk 4]
```diff
@@ -70,69 +81,103 @@ data Action = GenericHeaderActionController GenericHeader.Action
...
-      name = (fromMaybe "" profile.firstName) <> middleName <> lastName
-      gender = case (profile.gender) of
-        Just "MALE" -> Just MALE
-        Just "FEMALE" -> Just FEMALE
-        Just "OTHER" -> Just OTHER
-        Just "PREFER_NOT_TO_SAY" -> Just PREFER_NOT_TO_SAY
-        _ -> Nothing
-  continue state { data { name = name, editedName = name, gender = gender, emailId = profile.email} }
+
+eval (GenderSelected value) state = continue state{data{editedGender = Just value}, props{genderOptionExpanded = false , isBtnEnabled = (( length (trim state.data.editedName) >=3)  && state.props.isEmailValid)}}
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs:124]: (( length (trim state.data.editedName) >=3)

Comment (kavya-shree-s) [reply Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs:124]: Can be simplified too may brackets

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/View.purs [hunk 5]
```diff
@@ -117,101 +109,137 @@ view push state =
...
+                  [ height WRAP_CONTENT
+                  , width MATCH_PARENT
+                  , orientation VERTICAL
+                  ][  linearLayout
                       [ height WRAP_CONTENT
                       , width MATCH_PARENT
                       , orientation VERTICAL
-                      ]
-                      [ linearLayout
+                      , margin $ Margin 16 0 16 0
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/View.purs:182]: Use marginVertical

File: Frontend/ui-customer/src/Screens/OnBoardingFlow/PermissionScreen/View.purs [hunk 6]
```diff
@@ -27,7 +27,7 @@ import JBridge as JB
 import Language.Strings (getString)
 import Language.Types (STR(..))
 import Prelude (Unit, bind, const, pure, unit, (<<<), ($), (==), (<>), (/=),(&&))
-import PrestoDOM (Gravity(..), Length(..), Margin(..), Orientation(..), Padding(..), Visibility(..),PrestoDOM, Screen, afterRender, alignParentBottom, background, clickable, color, cornerRadius, fontStyle, gravity, height, imageView, imageWithFallback, lineHeight, linearLayout, margin, orientation, padding, text, textSize, textView, width,visibility)
+import PrestoDOM (Gravity(..), Length(..), Margin(..), Orientation(..), Padding(..), Visibility(..), Accessiblity(..), PrestoDOM, Screen, afterRender, alignParentBottom, background, clickable, color, cornerRadius, fontStyle, gravity, height, imageView, imageWithFallback, lineHeight, linearLayout, margin, orientation, padding, text, textSize, textView, width,visibility, id, accessibilityHint, accessibility)
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Screens/OnBoardingFlow/PermissionScreen/View.purs:30]: Unused imports

File: Frontend/ui-customer/src/Services/API.purs [hunk 7]
```diff
@@ -1954,4 +1971,35 @@ instance encodeRideFeedbackReq  :: Encode RideFeedbackReq where encode = default
...
-instance encodeRideFeedbackRes  :: Encode RideFeedbackRes where encode = defaultEncode
\ No newline at end of file
+instance encodeRideFeedbackRes  :: Encode RideFeedbackRes where encode = defaultEncode
+
+data GetDisabilityListReq = GetDisabilityListReq
+
+newtype GetDisabilityListResp = GetDisabilityListResp (Array Disability)
+
+-- data DisabilityType = BLIND_LOW_VISION |  HEAR_IMPAIRMENT | LOCOMOTOR_DISABILITY |  OTHER
+
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Services/API.purs:1981]: Remove comments code

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 8]
```diff
@@ -363,6 +363,22 @@ genderBannerConfig state =
...
+  let
+    config = Banner.config
+    config' = config
+      {
+        backgroundColor = Color.paleLavender
+      , title = (getString NOW_GET_ASSISTED_RIDES)
+      , titleColor = Color.purple
+      , actionText = (getString UPDATE_PROFILE)
+      , actionTextColor = Color.purple
+      , imageUrl = "ny_ic_accessibility_banner_img,https://assets.juspay.in/beckn/merchantcommon/images/ny_ic_banner_gender_feat.png"
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:377]: @MercyQueen Why gender banner image is used for accessibility , Please update
https://assets.juspay.in/beckn/merchantcommon/images/ny_ic_banner_gender_feat.png

File: Frontend/ui-customer/src/Services/Backend.purs [hunk 9]
```diff
@@ -17,44 +17,44 @@ module Services.Backend where
...
-import Services.Config as SC
-import Data.Lens ((^.))
-import Engineering.Helpers.Utils as EHU
-import Accessor (_deviceToken)
+import Debug(spy)
 
 getHeaders :: String -> Boolean -> Flow GlobalState Headers
 getHeaders val isGzipCompressionEnabled = do
     regToken <- loadS $ show REGISTERATION_TOKEN
+    _ <- pure $ spy "RegToken" regToken
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Services/Backend.purs:57]: Remove spy

File: Frontend/ui-customer/src/Strings/KN.js [hunk 10]
```diff
@@ -589,5 +589,21 @@ const kannadaStrings = {
   "JUST_ONE_LAST_THING" : "ಕೇವಲ ಒಂದು ಕೊನೆಯ ವಿಷಯ",
   "TOLL_CHARGES_WILL_BE_EXTRA" : "ಟೋಲ್/ಪಾರ್ಕಿಂಗ್ ಶುಲ್ಕಗಳು ಹೆಚ್ಚುವರಿಯಾಗಿರುತ್ತದೆ",
   "AUTO_RICKSHAW" : "ಆಟೋ ರಿಕ್ಷಾ",
-  "CABS_AVAILABLE" : "ಕ್ಯಾಬ್‌ಗಳು ಲಭ್ಯವಿದೆ"
+  "CABS_AVAILABLE" : "ಕ್ಯಾಬ್‌ಗಳು ಲಭ್ಯವಿದೆ",
+  "PHYSICAL_IMPAIRMENT_DESCRIPTION" : "&#8226;  ಚಾಲಕರಿಗೆ ನಿಮ್ಮ ನಿಶ್ಚಿತ ಸ್ಥಳಕ್ಕೆ ಪಿಕಪ್ ಗೆ ಬರುವಂತೆ ಸೂಚಿಸಲಾಗುತ್ತದೆ. <br> &#8226;  ಚಾಲಕರಿಗೆ ನಿಮ್ಮ ಗತಿಸಾಧನಾ ಉಪಕರಣದಲ್ಲಿ ನೆರವೇರಲು ಸೂಚಿಸಲಾಗುತ್ತದೆ. <br> &#8226;  ಚಾಲಕರಿಗೆ ನಿಮ್ಮ ಆವಶ್ಯಕತೆಗಳು ಮತ್ತು ಬೇಕಾದ ಕೋರಿಕೆಗಳನ್ನು ನೆನಪಿಗೆ ತರಲು ಸೂಚಿಸಲಾಗುತ್ತದೆ.",
+  "VISUAL_IMPAIRMENT_DESCRIPTION" : "&#8226;  ಚಾಲಕರಿಗೆ ಟೆಕ್ಸ್ಟ್ ಮಾಡುವ ಬದಲು ಕರೆಯುವಂತೆ ಸೂಚಿಸಲಾಗುತ್ತದೆ. <br> &#8226;  ಪಿಕಪ್ ಸಮಯದಲ್ಲಿ ಒಂದುಬಾರಿ ಹಾರ್ನ್ ಹೊಡೆಯುವಂತೆ ಸೂಚಿಸಲಾಗುತ್ತದೆ. <br> &#8226;  ಚಾಲಕರಿಗೆ ನಿಮ್ಮ ಆವಶ್ಯಕತೆಗಳು ಮತ್ತು ಬೇಕಾದ ಕೋರಿಕೆಗಳನ್ನು ನೆನಪಿಗೆ ತರಲು ಸೂಚಿಸಲಾಗುತ್ತದೆ.", 
+  "HEARING_IMPAIRMENT_DESCRIPTION" : "&#8226;  ಚಾಲಕರಿಗೆ ಕರೆಯುವಂತೆ ಟೆಕ್ಸ್ಟ್ ಮಾಡುವಂತೆ ಸೂಚಿಸಲಾಗುತ್ತದೆ. <br> &#8226;  ಪಿಕಪ್ ಸಮಯದಲ್ಲಿ ಒಂದುಬಾರಿ ನೀವು ಸಂದೇಶಿಸಲಾಗುತ್ತದೆ. <br> &#8226;  ಚಾಲಕರಿಗೆ ನಿಮ್ಮ ಆವಶ್ಯಕತೆಗಳು ಮತ್ತು ಬೇಕಾದ ಕೋರಿಕೆಗಳನ್ನು ನೆನಪಿಗೆ ತರಲು ಸೂಚಿಸಲಾಗುತ್ತದೆ.",
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Strings/KN.js:595]: "&#8226;  ಚಾಲಕರಿಗೆ ಕರೆಯುವಂತೆ ಟೆಕ್ಸ್ಟ್ ಮಾಡುವಂತೆ ಸೂಚಿಸಲಾಗುತ್ತದೆ. <br> &#8226; ಪಿಕಪ್‌ನಲ್ಲಿ ಒಮ್ಮೆ ನಿಮಗೆ ಸಂದೇಶ ಕಳುಹಿಸಲು ಚಾಲಕರನ್ನು ಪ್ರೇರೇಪಿಸಲಾಗುತ್ತದೆ;  ಚಾಲಕರಿಗೆ ನಿಮ್ಮ ಆವಶ್ಯಕತೆಗಳು ಮತ್ತು ಬೇಕಾದ ಕೋರಿಕೆಗಳನ್ನು ನೆನಪಿಗೆ ತರಲು ಸೂಚಿಸಲಾಗುತ್ತದೆ.",

Comment (kavya-shree-s) [reply Frontend/ui-customer/src/Strings/KN.js:595]: Replace

File: Frontend/ui-customer/src/Screens/OnBoardingFlow/AccountSetUpScreen/Transformer.purs [hunk 11]
```diff
@@ -0,0 +1,14 @@
+module Screens.AccountSetUpScreen.Transformer where
+
+import Prelude
+
+import Data.Array (foldl)
+import Screens.Types (DisabilityT)
+import Services.API (Disability(..)) --, DisabilityType(..))
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Screens/OnBoardingFlow/AccountSetUpScreen/Transformer.purs:7]: Remove comment line

**Comment by kavya-shree-s** [comment]
@MercyQueen Is it tested in IOS ?

File: Frontend/ui-customer/src/Components/CommonComponentConfig.purs [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        , config = config
+        , hideOthers = false
+        , topLeftIcon = true
+        , showBgColor = true
+        , editTextBgColor = Color.white900
+        , defaultText = Mb.fromMaybe "" disabilityData.otherDisabilityReason
+        }
+
+getDisabilityList = map \item ->
```
[the remark this answers is not in the document]
Comment (MercyQueen) [reply Frontend/ui-customer/src/Components/CommonComponentConfig.purs:133]: We are sending a variable to this already to take care of lazy evaluation

File: Frontend/ui-customer/src/Resources/Constants.purs (hunk 3 above)
Comment (MercyQueen) [reply Frontend/ui-customer/src/Resources/Constants.purs:307]: This is the default case and for other this is the id that is being used.

**Comment by MercyQueen** [comment]
> @MercyQueen Is it tested in IOS ?

yes

**Comment by kavya-shree-s** [comment]
@MercyQueen Can you squash the commits

File: Frontend/ui-common/src/Components/PopUpModal/View.purs [hunk 13]
```diff
@@ -161,11 +161,12 @@ view push state =
             , gravity $ state.secondaryText.gravity
             , padding state.secondaryText.padding
             , margin $ state.secondaryText.margin
-            , text $ state.secondaryText.text
+            , (text $ state.secondaryText.text)
```
Comment (kavya-shree-s) [inline Frontend/ui-common/src/Components/PopUpModal/View.purs:164]: @MercyQueen Remove extra brackets

File: Frontend/ui-common/src/Components/PopUpModal/View.purs [hunk 14]
```diff
@@ -317,6 +318,38 @@ view push state =
...
+    , orientation VERTICAL 
+    , margin $ Margin 16 0 16 20
+    ](map (\item-> 
+        linearLayout
+        [ height WRAP_CONTENT
+        , width MATCH_PARENT
+        , margin $ Margin 12 0 0 4
+        , orientation HORIZONTAL
+        ][  imageView
+            [ imageWithFallback $ "ny_ic_circle,https://assets.juspay.in/beckn/nammayatri/nammayatricommon/images/ny_ic_circle.png"
```
Comment (kavya-shree-s) [inline Frontend/ui-common/src/Components/PopUpModal/View.purs:335]: Can we get url from function

File: Frontend/ui-common/src/Components/PopUpModal/View.purs [hunk 15]
```diff
@@ -317,6 +318,38 @@ view push state =
...
+            [ width WRAP_CONTENT
+            , height WRAP_CONTENT
+            , margin $ MarginLeft 8
+            , text item 
+            , accessibility ENABLE
+            , accessibilityHint item
+            ] <> (FontStyle.getFontStyle FontStyle.ParagraphText LanguageStyle)
+
+        ]
+        ) (state.listViewArray))
```
Comment (kavya-shree-s) [inline Frontend/ui-common/src/Components/PopUpModal/View.purs:351]: remove extra brackets

File: Frontend/ui-customer/src/Screens/OnBoardingFlow/AccountSetUpScreen/View.purs [hunk 16]
```diff
@@ -78,26 +96,27 @@ view push state =
...
                         , weight 1.0
                         , orientation VERTICAL
-                        ]
-                        []
+                        ][]
+                    , if state.data.editedDisabilityOptions.activeIndex == 1 && (isJust state.data.disabilityOptions.selectedDisability) then disabilityClaimerView state push  else textView[]
                     ]
                 ]
             ]
+        -- , if state.data.disabilityOptions.activeIndex == 1 then disabilityClaimerView state push  else textView[]
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Screens/OnBoardingFlow/AccountSetUpScreen/View.purs:119]: Remove

File: Frontend/ui-common/src/Components/SelectListModal/View.purs (hunk 2 above)
Comment (kavya-shree-s) [reply Frontend/ui-common/src/Components/SelectListModal/View.purs:149]: @MercyQueen Change this

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**